geoapps/io/input_file.py

```diff
@@ -401,3 +391,104 @@ def _set_associations(self, d: dict[str, Any]) -> None:
 
             else:
                 continue
+
+
+class UIJson:
+    """Encodes the ui.json format and provides utilities."""
+
+    def __init__(self, ui: dict[str, Any]):
+        self.ui = ui
+
+    @staticmethod
+    def _data(d: dict[str, Any]) -> dict[str, Any]:
+        """Flattens ui.json format to simple key/value pair."""
+        data = {}
+        for k, v in d.items():
+            if isinstance(v, dict):
+                field = "value" if UIJson._truth(d, k, "isValue") else "property"
+                if not UIJson._truth(d, k, "enabled"):
+                    data[k] = None
+                else:
+                    data[k] = v[field]
+            else:
+                data[k] = v
+
+        return data
+
+    def data(self):
+        """Applies _data to self.ui."""
+        return UIJson._data(self.ui)
+
+    @staticmethod
+    def _collect(d: dict[str, Any], field: str, value: Any = None) -> dict[str, Any]:
+        """Collects ui parameters with common field and optional value."""
+        data = {}
+        for k, v in d.items():
+            if isinstance(v, dict):
+                if field in v.keys():
+                    if value is None:
+                        data[k] = v
+                    else:
+                        if v[field] == value:
+                            data[k] = v
+        return data if data else None
+
+    def collect(self, field: str, value: Any = None) -> dict[str, Any]:
+        """Applies _collect to self.ui."""
+        return UIJson._collect(self.ui, field, value)
```

Comment:
same here, looks like we are going to loop again everytime we ask for the `collect`

Reply:
In this case, I think this would be the correct behaviour.  This method just collects and returns groups of parameters.  What you're describing would be like a 'filter' method - could also be useful

Reply:
gotcha